solar_consumer/data/fetch_de_data.py

```diff
@@ -0,0 +1,90 @@
+import os
+import pandas as pd
+import dotenv
+from datetime import datetime, timedelta, timezone
+import requests
+import xml.etree.ElementTree as ET
+from loguru import logger
+
+# Load environment variables
+dotenv.load_dotenv()
+
+def fetch_de_data(historic_or_forecast: str = "generation") -> pd.DataFrame:
+    """
+    Fetch solar generation data from German bidding zones via the
+    ENTSOE API
+
+    Only 'generation' mode is supported for now
+    
+    Returns DataFrame with 3 columns:
+      - target_datetime_utc (UTC date and time)
+      - solar_generation_kw (generation in kilowatts)
+      - tso_zone (bidding zone code)
+    """
+    
+    assert historic_or_forecast == "generation", "Only 'generation' supported for the time being"
+
+    # Fetches data from last 24 hours from current time
+    now = datetime.now(timezone.utc)
+    start = now - timedelta(hours=24)
+    period_start = start.strftime("%Y%m%d%H%M")
+    period_end = now.strftime("%Y%m%d%H%M")
+
+    # Prepare request
+    url = "https://web-api.tp.entsoe.eu/api" # base url for api
+    api_key = os.getenv("ENTSOE_API_KEY", "") # api key from env vars, empty string if missing
+    params = {
+        "documentType": "A75",    # actual generation
+        "processType": "A16",     # realised output
+        "periodStart": period_start,
+        "periodEnd": period_end,
+        "securityToken": api_key,
+    }
+
```

Comment:
Is it worth using something like this - https://github.com/EnergieID/entsoe-py, or perhaps this hasnt been updated and might introduce more risk